src/main/java/teammates/ui/controller/InstructorCoursesPageData.java

```diff
@@ -1,35 +1,192 @@
 package teammates.ui.controller;
 
-import java.util.HashMap;
+import java.util.ArrayList;
 import java.util.List;
 
 import teammates.common.datatransfer.AccountAttributes;
 import teammates.common.datatransfer.InstructorAttributes;
 import teammates.common.datatransfer.CourseAttributes;
 import teammates.common.datatransfer.CourseDetailsBundle;
+import teammates.common.util.Const;
+import teammates.common.util.Sanitizer;
+import teammates.logic.api.Logic;
+import teammates.ui.template.ActiveCoursesTable;
+import teammates.ui.template.ActiveCoursesTableRow;
+import teammates.ui.template.ArchivedCoursesTable;
+import teammates.ui.template.ArchivedCoursesTableRow;
+import teammates.ui.template.ElementTag;
 
 /**
  * This is the PageData object for the 'Courses' page 
  */
 public class InstructorCoursesPageData extends PageData {
+    private ArchivedCoursesTable archivedCourses;
+    private ActiveCoursesTable activeCourses;
+    private String courseIdToShow;
+    private String courseNameToShow;
     
     public InstructorCoursesPageData(AccountAttributes account) {
         super(account);
     }
     
-    /** Used when adding a course. Null if not adding a course. */
-    public CourseAttributes newCourse;
+    public void init(ArrayList<CourseDetailsBundle> allCourses){
+        init(allCourses, "", ""); 
+    }
+    
+    public void init(ArrayList<CourseDetailsBundle> allCourses, String courseIdToShowParam, String courseNameToShowParam){
+        activeCourses = extractActiveCourses(allCourses);
+        archivedCourses = extractArchivedCourses(allCourses);
+        courseIdToShow = courseIdToShowParam;
+        courseNameToShow = courseNameToShowParam;
+    }
+    
+    public String getCourseIdToShow() {
+        return courseIdToShow;
+    }
     
-    public HashMap<String, InstructorAttributes> instructors;
+    public String getCourseNameToShow() {
+        return courseNameToShow;
+    }
+
+    public ActiveCoursesTable getActiveCourses() {
+        return activeCourses;
+    }
+
+    public ArchivedCoursesTable getArchivedCourses() {
+        return archivedCourses;
+    }
     
-    /* List of details for all courses created by an instructor */
-    public List<CourseDetailsBundle> allCourses;
-    /* List of all archived courses from an instructor */
-    public List<CourseAttributes> archivedCourses;
+    private ArchivedCoursesTable extractArchivedCourses(List<CourseDetailsBundle> courseBundles) {
+        ArchivedCoursesTable archivedCourses = new ArchivedCoursesTable();
+        Logic logic = new Logic();
+
+        CourseDetailsBundle.sortDetailedCoursesByCourseId(courseBundles);
```

Comment:
If you are only using Logic to check for the archive status, you can use the instructor's isArchived attribute. CourseAttribute's isArchived is not relevant anymore (see #3511)